server/src/test/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalVariableWidthHistogramTests.java

```diff
@@ -0,0 +1,353 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.bucket.histogram;
+
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.MockBigArrays;
+import org.elasticsearch.common.util.MockPageCacheRecycler;
+import org.elasticsearch.indices.breaker.NoneCircuitBreakerService;
+import org.elasticsearch.script.ScriptService;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.aggregations.InternalAggregation;
+import org.elasticsearch.search.aggregations.InternalAggregations;
+import org.elasticsearch.search.aggregations.MultiBucketConsumerService;
+import org.elasticsearch.search.aggregations.ParsedMultiBucketAggregation;
+import org.elasticsearch.search.aggregations.pipeline.PipelineAggregator;
+import org.elasticsearch.test.InternalMultiBucketAggregationTestCase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class InternalVariableWidthHistogramTests extends
+    InternalMultiBucketAggregationTestCase<InternalVariableWidthHistogram>{
+
+    private DocValueFormat format;
+    private InternalVariableWidthHistogram.EmptyBucketInfo emptyBucktInfo;
+        private int numBuckets;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        format = randomNumericDocValueFormat();
+        emptyBucktInfo = new InternalVariableWidthHistogram.EmptyBucketInfo(InternalAggregations.EMPTY);
+        this.numBuckets = 3;
+    }
+
+    @Override
+    protected InternalVariableWidthHistogram createTestInstance(String name,
+                                                                Map<String, Object> metaData,
+                                                                InternalAggregations aggregations) {
+        List<InternalVariableWidthHistogram.Bucket> buckets = new ArrayList<>();
```

Comment:
Its more normal for these test classes to make random buckets too. That way you get tests for the bucket serialization.

Reply:
I updated `createTestInstance()` to create actual test instances, instead of empty ones.

This enabled some of the `InternalAggregationTestCase` tests that weren't doing anything on the empty test instances from earlier, to run. They exposed some bugs, such as how I had been updating `reduceContext.consumeBucketsAndMaybeBreak()` incorrectly (fixed now).